cloudify/decorators.py

```diff
@@ -40,4 +45,87 @@ def workflow(func=None, system_wide=False, resumable=False, **kwargs):
                                    resumable=resumable, **kwargs)
 
 
+def serial_operation(revert_to_parallel_threshold=0,
+                     workflows=None,
+                     states=None):
+    """Control order of multiple node instances of a single node template.
+        Normally, multiple node instance operations will be
+        executed in parallel.
+        However, some cases, like clusters require that cluster members are
+        installed serially.
+
+    :param revert_to_parallel_threshold: Switch back to parallel execution
+        after n node instances. Integer.
+        0: Never use parallel. Always straight serial.
+        1: After 1 node is installed, return to Parallel. Default.
+        n + 1: After n node instances.
+    :param workflows: List of workflows to apply serialization
+        to. The default is ['install']. This means we scale will
+        execute in parallel.
+    :param states: List of states that the preceding
+        node instance may be in before executing the current node instance.
+        The default is ['started']. We will only continue to next instance
+        after the preceding instance is in "started" state.
+    :raises: CloudifySerializationRetry.
+    :return: decorator wrapper
+    """
+
+    workflows = workflows if isinstance(
+        workflows, list) else ['install']
+    states = states if isinstance(
+        states, list) else ['started']
```

Comment:
just ignoring the passed-in value if it's of the wrong type, is pretty bad api

Reply:
will change it.